editor/plugins/path_2d_editor_plugin.cpp

```diff
@@ -475,9 +485,28 @@ void Path2DEditor::_mode_selected(int p_mode) {
 		curve_edit->set_pressed(false);
 		curve_edit_curve->set_pressed(false);
 		curve_del->set_pressed(true);
-	} else if (p_mode == ACTION_CLOSE) {
-		//?
+	} else if (p_mode == MODE_CLEAR_POINTS) {
+		if (!node->get_curve().is_valid()) {
+			return;
+		}
+		if (node->get_curve()->get_point_count() == 0) {
+			return;
+		}
+		EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+		int history_id = undo_redo->get_history_id_for_object(node); // Uses the same history as the selected node instead of a global history.
+
+		if (history_id != EditorUndoRedoManager::INVALID_HISTORY) {
+			PackedVector2Array points = node->get_curve()->get_points().duplicate();
 
+			undo_redo->create_action_for_history(TTR("Clear Curve Points"), history_id);
```

Comment:
```suggestion
			undo_redo->create_action(TTR("Clear Curve Points"), node);
```
`create_action_for_history()` is needed only for very specific cases.